sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_ops_helper.py

```diff
@@ -174,29 +174,49 @@ def list_logs(run_operations: RunOperations, job_resource: JobBase) -> Dict:
     return {key: logs_dict[key] for key in keys}
 
 
+def _coerce_data_type(v: Optional[str]) -> Optional[str]:
+    return v and str(v).lower().replace("_", "")
+
+
 # pylint: disable=too-many-statements,too-many-locals
 def stream_logs_until_completion(
     run_operations: RunOperations,
     job_resource: JobBase,
     datastore_operations: Optional[DatastoreOperations] = None,
     raise_exception_on_failed_job: bool = True,
     *,
-    requests_pipeline: HttpPipeline
+    requests_pipeline: HttpPipeline,
+    file_handle: Optional[TextIO] = None,
+    from_start: bool = False
 ) -> None:
```

Comment:
The code currently checks multiple times for the existence of datastore_operations and the outputs property in job_resource. These conditions can be combined to reduce redundancy and improve efficiency.
Consider using early returns for error conditions to avoid deep nesting. For example, if datastore_operations is None or outputs does not exist, return early from the function.